src/combat.cpp

```diff
@@ -608,21 +608,22 @@ void Combat::combatTileEffects(const SpectatorVec& spectators, Creature* caster,
 	}
 }
 
-void Combat::postCombatEffects(Creature* caster, const Position& pos, const CombatParams& params)
+void Combat::postCombatEffects(std::shared_ptr<Creature> caster, const Position& pos, const CombatParams& params)
 {
 	if (caster && params.distanceEffect != CONST_ANI_NONE) {
 		addDistanceEffect(caster, caster->getPosition(), pos, params.distanceEffect);
 	}
 }
 
-void Combat::addDistanceEffect(Creature* caster, const Position& fromPos, const Position& toPos, uint8_t effect)
+void Combat::addDistanceEffect(std::shared_ptr<Creature> caster, const Position& fromPos, const Position& toPos,
+                               uint8_t effect)
 {
 	if (effect == CONST_ANI_WEAPONTYPE) {
 		if (!caster) {
 			return;
 		}
 
-		Player* player = caster->getPlayer();
+		auto player = caster->getPlayer();
```

Comment:
Could this be const auto
```suggestion
		const auto player = caster->getPlayer();
```